===== Patch instructions generated by SAVER =====
[+] { Move free(*(read_crypto_list:cur)) at 16 (line 150, column 4) to 15 (line 151, column 4) under true }
===== Actual patch in diff format =====
@@ -147,8 +147,8 @@ read_crypto_list (const char *prefix)
       if (! cur->modname)
 	{
 	  grub_errno = GRUB_ERR_NONE;
-	  grub_free (cur);
 	  grub_free (cur->name);
+	  grub_free (cur);
 	  continue;
 	}
       cur->next = crypto_specs;
